unit_tests/utilities/test_zaza_utilities_openstack.py

```diff
@@ -814,14 +813,16 @@ def test_get_current_openstack_release_pair(self):
             name='_get_machine_series'
         )
 
+        _machine = mock.MagicMock()
+
         # No machine returned
-        self._get_machines.side_effect = StopIteration
+        self._get_machines.side_effect = KeyError
```

Comment:
This is not really testing the thing you want to test?  I think it would be better to say:

```python
self._get_machines.return_value = []
```

And that means that the test is testing that the target code will handle the empty list, rather than make the code implementation dependent on the test raising `KeyError`?